### PR TITLE
Alter table analyzer cleanup

### PR DESCRIPTION
**Pull request by mfussenegger**
See commit messages

### CODE REVIEW

File: sql/src/main/java/io/crate/analyze/AlterTableAnalyzer.java [hunk 1]
```diff
@@ -21,82 +21,88 @@
...
-    }
-
-    public AnalyzedStatement analyze(Node node, Analysis analysis) {
-        return process(node, analysis);
+        PartitionName partitionName = PartitionPropertiesAnalyzer.toPartitionName(
+            tableInfo,
+            partitionsProperties,
+            parameters
+        );
+        if (tableInfo.partitions().contains(partitionName) == false) {
```
Comment (matriv) [inline sql/src/main/java/io/crate/analyze/AlterTableAnalyzer.java]: The ! converted to == false was intentional?

File: sql/src/main/java/io/crate/analyze/AlterBlobTableAnalyzer.java [hunk 2]
```diff
@@ -21,34 +21,39 @@
...
+                tableParameter,
+                tableInfo.tableParameterInfo(),
+                node.genericProperties(),
+                parameters
+            );
         } else if (!node.resetProperties().isEmpty()) {
-            TABLE_PROPERTIES_ANALYZER.analyze(
-                statement.tableParameter(), statement.table().tableParameterInfo(),
-                node.resetProperties());
+            TablePropertiesAnalyzer.analyze( tableParameter, tableInfo.tableParameterInfo(), node.resetProperties());
```
Comment (matriv) [inline sql/src/main/java/io/crate/analyze/AlterBlobTableAnalyzer.java]: space in `analyze( tableParameter`

File: sql/src/main/java/io/crate/analyze/AlterTableAnalyzer.java (hunk 1 above)
Comment (mfussenegger) [inline sql/src/main/java/io/crate/analyze/AlterTableAnalyzer.java]: yes

File: sql/src/main/java/io/crate/analyze/AlterTableAnalyzer.java [hunk 3]
```diff
@@ -21,82 +21,88 @@
...
-    private static final TablePropertiesAnalyzer TABLE_PROPERTIES_ANALYZER = new TablePropertiesAnalyzer();
     private final Schemas schemas;
 
     AlterTableAnalyzer(Schemas schemas) {
         this.schemas = schemas;
     }
 
-    @Override
-    public AlterTableAnalyzedStatement visitColumnDefinition(ColumnDefinition node, Analysis analysis) {
-        if (node.ident().startsWith("_")) {
```
Comment (matriv) [inline sql/src/main/java/io/crate/analyze/AlterTableAnalyzer.java]: Why do we remove this check?

Comment (mfussenegger) [inline sql/src/main/java/io/crate/analyze/AlterTableAnalyzer.java]: Was unused

**Pull request merged**